### PR TITLE
Use `get_by_natural_key` for `PasswordResetEmail`

### PR DESCRIPTION
**Pull request by KevinEtienne**

### CODE REVIEW

**Comment by KevinEtienne** [comment]
@Ian-Foote can you review/merge?

File: user_management/api/views.py [hunk 1]
```diff
@@ -124,7 +124,7 @@ def post(self, request, *args, **kwargs):
 
         email = serializer.data['email']
         try:
-            user = User.objects.get(email__iexact=email)
+            user = User.objects.get_by_natural_key(email)
```
Comment (Ian-Foote) [inline user_management/api/views.py]: Do we have a default implementation of `get_by_natural_key`? Does it preseve the `iexact` lookup?

Comment (KevinEtienne) [inline user_management/api/views.py]: Well spotted, looks like we didn't test `iexact`.

**Comment by Ian-Foote** [comment]
Happy to merge when travis is.

**Pull request merged**